packages/shared/src/components/squads/Details.tsx

```diff
@@ -28,15 +29,24 @@ import { SquadPrivacyState } from './common/SquadPrivacyState';
 import { SquadDangerZone } from './settings/SquadDangerZone';
 import { Squad } from '../../graphql/sources';
 import { useViewSize, ViewSize } from '../../hooks';
+import { useSlackChannelsQuery } from '../../hooks/integrations/slack/useSlackChannelsQuery';
+import { Dropdown } from '../fields/Dropdown';
+import { Typography, TypographyType } from '../typography/Typography';
 
 const squadImageId = 'squad_image_file';
 
 interface SquadDetailsProps {
-  onSubmit: (e: FormEvent, formJson: SquadForm) => void;
+  onSubmit: (
+    e: FormEvent,
+    formJson: SquadForm,
+    selectedChannel?: string,
+  ) => void;
   onRequestClose?: () => void;
   children?: ReactNode;
   isLoading?: boolean;
   squad?: Squad;
+  initialData?: Partial<Squad>;
```

Comment:
Just to be safe and so the spread won't break the app.
```suggestion
  initialData?: Partial<Squad> = {};
```